rewrite-maven/src/main/java/org/openrewrite/maven/ChangePluginConfiguration.java

```diff
@@ -74,24 +91,45 @@ public Xml visitTag(Xml.Tag tag, ExecutionContext ctx) {
                     Optional<Xml.Tag> maybePlugin = plugins.getChildren().stream()
                             .filter(plugin ->
                                     "plugin".equals(plugin.getName()) &&
-                                            groupId.equals(plugin.getChildValue("groupId").orElse(null)) &&
-                                            artifactId.equals(plugin.getChildValue("artifactId").orElse(null))
+                                    groupId.equals(plugin.getChildValue("groupId").orElse(null)) &&
+                                    artifactId.equals(plugin.getChildValue("artifactId").orElse(null))
                             )
                             .findAny();
                     if (maybePlugin.isPresent()) {
                         Xml.Tag plugin = maybePlugin.get();
-                        if (configuration == null) {
+                        if (configuration == null && xslt == null && xsltResource == null) {
                             plugins = filterChildren(plugins, plugin,
                                     child -> !(child instanceof Xml.Tag && "configuration".equals(((Xml.Tag) child).getName())));
-                        } else {
+                        } else if (configuration != null && xslt == null && xsltResource == null) {
```

Comment:
I'm wondering if we should simplify the logic here such that it's easier to follow and extend. Right now when there's both a configuration and xslt supplied, we don't do anything. That's perhaps confusing to folks. Instead I propose we let configuration to precedence, and update the logic to reflect that. Would you agree to that?
```suggestion
                        } else if (configuration != null) {
```

Reply:
Yes, that's a great idea. I had a clone of the ChangePluginConfiguration doing only the XSLT transformations on my branch.